### ISSUE TITLE
🗑️ Deprecation promotion of `toPositiveInt` function in `ResultContext` to error level

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#464]
## 📝 Description

We would like to promote the deprecation of the `toPositiveInt` function in `ResultContext` to an error level.

## 🔗 Dependencies

This issue is blocked by the following ones:
- [x] #463

## ✅ Checklist

- [ ] Promote the deprecation of this declaration, remove its tests and update the [unreleased changelog].
- [ ] Close this issue as completed and update tracking ones if present.

[unreleased changelog]: https://github.com/kotools/types/blob/main/CHANGELOG.md#unreleased

**Comment by LVMVRQUXL** [comment]
Another issue for this task will be created when ready.
